taxi/forms.py

```diff
@@ -0,0 +1,54 @@
+from django import forms
+from django.contrib.auth.forms import UserCreationForm
+
+from taxi.models import Driver, Car
+
+
+class LicenseNumberValidationMixin(forms.ModelForm):
+    LENGTH = 8
+    FIRST_LETTERS = 3
+    LAST_DIGITS = 5
+
+    def clean_license_number(self) -> str:
+        license_number = self.cleaned_data["license_number"]
+        if len(license_number) != self.LENGTH:
+            raise forms.ValidationError(
+                f"License number length should be {self.LENGTH} characters"
+            )
+        if (
+            not license_number[: self.FIRST_LETTERS].isupper()
+            or not license_number[: self.FIRST_LETTERS].isalpha()
+        ):
+            raise forms.ValidationError(
+                f"First {self.FIRST_LETTERS} letters should be uppercase"
+            )
+        if not license_number[self.FIRST_LETTERS:].isdigit():
+            raise forms.ValidationError(
+                f"Last {self.LAST_DIGITS} characters should be digits"
+            )
+        return license_number
+
+
+class DriverLicenseUpdateForm(LicenseNumberValidationMixin, forms.ModelForm):
+    class Meta:
+        model = Driver
```

Comment:
check if your homework is according to checklis[https://github.com/mate-academy/py-taxi-service-user-form-and-validation/blob/master/checklist.md]